digital_benchmark/accounts/views.py

```diff
@@ -51,9 +53,39 @@ def post(self, request):
                 return redirect(next)
             return redirect('/')
 
-        return render(request,'signup.html',{'form':form})
+        return render(request, 'signup.html', {'form': form})
+
 
 class LogoutView(View):
     def get(self, request):
         logout(request)
-        return redirect('/')
\ No newline at end of file
+        return redirect('/')
+
+
+class Signup(APIView):
+    def post(self, request, format=None):
+        try:
+            user = User.objects.get(username=request.data.get('username', ''))
```

Comment:
Validate api input params through serializer. Also add UniqueValidator for 'username' field.